itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/WarehouseInstance.java

```diff
@@ -203,25 +201,14 @@ private Path createAvroSchemaFile(FileSystem fs, Path testPath) throws IOExcepti
             "  \"tableName\" : \"table1\"",
             "}"
     };
-    createTestDataFile(schemaFile.toUri().getPath(), schemaVals);
-    return schemaFile;
-  }
 
-  private void createTestDataFile(String filename, String[] lines) throws IOException {
-    FileWriter writer = null;
-    try {
-      File file = new File(filename);
-      file.deleteOnExit();
-      writer = new FileWriter(file);
-      int i=0;
-      for (String line : lines) {
-        writer.write(line + "\n");
-      }
-    } finally {
-      if (writer != null) {
-        writer.close();
+    try (FSDataOutputStream stream = fs.create(schemaFile)) {
+      for (String line : schemaVals) {
+        stream.write((line + "\n").getBytes());
       }
     }
+    fs.deleteOnExit(schemaFile);
+    return schemaFile;
```

Comment:
Not sure why this comment got missed but adding it again.

Identifying the changes separately is always not enough since you have to do one round with existing declaration/definition of classes and then another with the required change. It seems to be a lot overhead for the developer. Additionally refactoring  tasks are hardly of importance and this seems to be more so for hive since there is lot of existing code and no one wants to touch it to correct the design. 
Doing refactoring at the same time also allows developers to keep things in context and iterate over changes as he is doing it rather than sit back and design a new change. Also for the reviewer i think as much attention has to be paid for refactored code vs new code. I